util/dvsim/OneShotCfg.py

```diff
@@ -0,0 +1,187 @@
+            "F": self.scratch_path + "/" + "failed",
+            "K": self.scratch_path + "/" + "killed"
+        }
+
+        # Use the default build mode for tests that do not specify it
+        if not hasattr(self, "build_mode"):
+            setattr(self, "build_mode", "default")
+
+        self._process_exports()
+
+        # Create objects from raw dicts - build_modes, sim_modes, run_modes,
+        # tests and regressions, only if not a master cfg obj
+        if not self.is_master_cfg:
+            self._create_objects()
+
+        # Post init checks
+        self.__post_init__()
+
+    def __post_init__(self):
+        # Run some post init checks
+        super().__post_init__()
+
+    @staticmethod
+    def create_instance(flow_cfg_file, proj_root, args):
+        '''Create a new instance of this class as with given parameters.
+        '''
+        return OneShotCfg(flow_cfg_file, proj_root, args)
+
+    # Purge the output directories. This operates on self.
+    def _purge(self):
+        if self.scratch_path is not "":
+            try:
+                log.info("Purging scratch path %s", self.scratch_path)
+                os.system("/bin/rm -rf " + self.scratch_path)
+            except IOError:
+                log.error('Failed to purge scratch directory %s',
+                          self.scratch_path)
+
+    def _create_objects(self):
+        # Create build and run modes objects
+        build_modes = Modes.create_modes(BuildModes,
+                                         getattr(self, "build_modes"))
+        setattr(self, "build_modes", build_modes)
+
+        # All defined build modes are being built, h
+        # ence extend all with the global opts.
+        for build_mode in build_modes:
+            build_mode.build_opts.extend(self.build_opts)
+
+    def _print_list(self):
+        for list_item in self.list_items:
+            log.info("---- List of %s in %s ----", list_item, self.name)
+            if hasattr(self, list_item):
+                items = getattr(self, list_item)
+                for item in items:
+                    log.info(item)
+            else:
+                log.error("Item %s does not exist!", list_item)
+
+    def _create_dirs(self):
+        '''Create initial set of directories
+        '''
+        # Invoking system calls has a performance penalty.
+        # Construct a single command line chained with '&&' to invoke
+        # the system call only once, rather than multiple times.
+        create_link_dirs_cmd = ""
+        for link in self.links.keys():
+            create_link_dirs_cmd += "/bin/rm -rf " + self.links[link] + " && "
+            create_link_dirs_cmd += "mkdir -p " + self.links[link] + " && "
+        create_link_dirs_cmd += " true"
+
+        try:
+            os.system(create_link_dirs_cmd)
+        except IOError:
+            log.error("Error running when running the cmd \"%s\"",
+                      create_link_dirs_cmd)
+            sys.exit(1)
```

Comment:
Looks like this is identical to SimCfg::_create_dirs(). Perhaps we can reuse (I will clean this up later).